pkg/command/service/clean_test.go

```diff
@@ -0,0 +1,113 @@
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package service
+
+import (
+	"testing"
+
+	"gotest.tools/assert"
+	corev1 "k8s.io/api/core/v1"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	k8sfake "k8s.io/client-go/kubernetes/fake"
+	"knative.dev/kperf/pkg"
+	"knative.dev/kperf/pkg/testutil"
+	servingv1client "knative.dev/serving/pkg/client/clientset/versioned/typed/serving/v1"
+	servingv1fake "knative.dev/serving/pkg/client/clientset/versioned/typed/serving/v1/fake"
+)
+
+func TestNewServiceCleanCommand(t *testing.T) {
+	t.Run("incompleted or wrong args for service clean", func(t *testing.T) {
+		client := k8sfake.NewSimpleClientset()
+		fakeServing := &servingv1fake.FakeServingV1{Fake: &client.Fake}
+		servingClient := func() (servingv1client.ServingV1Interface, error) {
+			return fakeServing, nil
+		}
+
+		p := &pkg.PerfParams{
+			ClientSet:        client,
+			NewServingClient: servingClient,
+		}
+		cmd := NewServiceCleanCommand(p)
+
+		_, err := testutil.ExecuteCommand(cmd)
+		assert.ErrorContains(t, err, "both namespace and namespace-prefix are empty")
+
+		_, err = testutil.ExecuteCommand(cmd, "--namespace-prefix", "test-kperf-1", "--namespace-range", "2,1")
+		assert.ErrorContains(t, err, "failed to parse namespace range 2,1")
+
+		_, err = testutil.ExecuteCommand(cmd, "--namespace-prefix", "test-kperf", "--namespace-range", "x,y")
+		assert.ErrorContains(t, err, "strconv.Atoi: parsing \"x\": invalid syntax")
+
+		_, err = testutil.ExecuteCommand(cmd, "--namespace-prefix", "test-kperf", "--namespace-range", "1,y")
+		assert.ErrorContains(t, err, "strconv.Atoi: parsing \"y\": invalid syntax")
+
+		_, err = testutil.ExecuteCommand(cmd, "--namespace-prefix", "test-kperf", "--namespace-range", "1")
+		assert.ErrorContains(t, err, "expected range like 1,500, given 1")
+	})
+
+	t.Run("clean service as expected", func(t *testing.T) {
+		ns := &corev1.Namespace{
+			ObjectMeta: metav1.ObjectMeta{
+				Name: "test-kperf-1",
+			},
+		}
+		client := k8sfake.NewSimpleClientset(ns)
+		fakeServing := &servingv1fake.FakeServingV1{Fake: &client.Fake}
+		servingClient := func() (servingv1client.ServingV1Interface, error) {
+			return fakeServing, nil
+		}
+
+		p := &pkg.PerfParams{
+			ClientSet:        client,
+			NewServingClient: servingClient,
+		}
+
+		cmd := NewServiceCleanCommand(p)
+		_, err := testutil.ExecuteCommand(cmd, "--namespace", "test-kperf-1")
+		assert.NilError(t, err)
+
+		_, err = testutil.ExecuteCommand(cmd, "--namespace-prefix", "test-kperf", "--namespace-range", "1,2")
+		assert.NilError(t, err)
+
+		_, err = testutil.ExecuteCommand(cmd, "--namespace-prefix", "test-kperf", "--namespace-range", "x,y")
```

Comment:
This case is duplicated with the third case in the `incompleted or wrong args for service clean`.